File: sublime/link.jl
link_target_path = "$(@__DIR__)/User"
if Sys.iswindows()
    config_path = "$(ENV["USERPROFILE"])/scoop/persist/sublime-text/Data/Packages/User"
elseif Sys.islinux()
    config_path = "$(ENV["HOME"])/.config/sublime-text/Packages/User"
elseif Sys.isapple()
    config_path = "$(ENV["HOME"])/Library/Application Support/Sublime Text/Packages/User"
else
    println("OS not yet supported")
    exit()
end

if ispath(config_path)
    print("Config already exists, overwrite? [y/N]: ")
    answer = readline()
    if answer == "y" || answer == "Y"
        rm(config_path, recursive=true)
    else
        exit()
    end
end

symlink(link_target_path, config_path, dir_target=true)
println("Symlink created")

File: sublime/link.py
#!/usr/bin/env python3

from sys import path
from os.path import realpath, dirname
path.append(dirname(dirname(realpath(__file__))))
from utils import *

check_os(OS.LINUX, OS.WINDOWS, OS.MACOS)

target = join(target_dir(), "User")

match platform:
    case OS.LINUX:
        link = join(DOTCONFIG, "sublime-text", "Packages", "User")
    case OS.MACOS:
        link = join(APPLICATION_SUPPORT, "Sublime Text", "Packages", "User")
    case OS.WINDOWS:
        if check_scoop():
            link = join(USERPROFILE, "scoop", "persist", "sublime-text", "Data", "Packages", "User")
        else:
            link = join(APPDATA, "Sublime Text", "Packages", "User")

make_symlink(target, link)
